### DISCUSSION TITLE
Providing registry username and password for Azure Container Instances

### DISCUSSION THREAD

**Discussion by mickvikt** [nuke-build/nuke#868] [Q&A] [answered]
Hello,
I'm trying to create a target to deploy a Docker container to Azure Container Instances. I'm thinking of using path executable for the Azure CLI az command.

Since the Docker image is uploaded to a private Azure container registry, I need to provide the registry password. Here's the syntax of the command:

`az container create -g MyResourceGroup --name myapp --image myAcrRegistry.azurecr.io/myimage:latest --registry-password password
`
I can use the Secret attribute and `nuke :secrets` to provide this password as a secret, but when I run the Azure CLI command via the path executable, my resolved secret is displayed in the console where the invoked command is being displayed with all of its arguments.

Can this be prevented?
Is there a better way to provide credentials for the CLI tools?

Thank you.

**Comment by matkoch** [comment, marked as answer]
The `Tool` delegate has an `outputFilter` parameter. Might convert this to a "sanatize list" later, but as of know you can just replace your secret there. Alternatively, hide the invocation via `logInvocation: false`.

**Reply by mickvikt** [reply]
Thanks, Matthias.
It does make sense. I've overlooked the outputFilter parameter.
